src/coreclr/src/tools/aot/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -1280,13 +1280,20 @@ private static bool IsTypeSpecForTypicalInstantiation(TypeDesc t)
             {
                 bool devirt;
 
+                // Check For interfaces before the bubble check
+                // since interface methods 
+                if (targetMethod.OwningType.IsInterface)
+                {
+                    // Handle interface methods specially because the Sealed bit has no meaning on interfaces.
+                    devirt = !targetMethod.IsVirtual;
+                }
                 // if we are generating version resilient code
                 // AND
                 //    caller/callee are in different version bubbles
                 // we have to apply more restrictive rules
                 // These rules are related to the "inlining rules" as far as the
                 // boundaries of a version bubble are concerned.
-                if (!_compilation.NodeFactory.CompilationModuleGroup.VersionsWithMethodBody(callerMethod) ||
+                else if (!_compilation.NodeFactory.CompilationModuleGroup.VersionsWithMethodBody(callerMethod) ||
```

Comment:
though this change does work for this case, need to ensure whether the bubble check should ignore methods in the same module. Wonder if there are other cases falling into this path incorrectly.

Reply:
Mmm.. I suspect the real check here should be that the typical method versions with the method body, instead of checking the instantiated method.

Reply:
Yeah, it seems to me that `VersionWithMethodBody` should return true in this case. That's the real difference between CG1 and CG2. CG1 considers two types in the same version bubble if their assemblies are in the same bubble, or if the NonVersionableTypeAttribute is present on the type. CG2 checks a method cache that it builds with a bunch of new code vs CG1.

Reply:
hmm, that was my initial thought as well. Will update the fix. Would be good to add clarity why VersionWIthMethodBody can be utilized for bubble checks.

Reply:
updated now.